operation-and-usage/operations/nso-device-manager.md

```diff
@@ -952,6 +976,57 @@ Authenticating southbound using stored configuration has two main components to
 2. Public key. This means that a private key, either from a file in the user's SSH key directory, or one that is configured in the /ssh/private-key list in the NSO configuration, is used for authentication. Refer to [Publickey Authentication](ssh-key-management.md#d5e4113) for the details on how the private key is selected.
 3. Finally, an interesting option is to use the 'same-pass' option. Since NSO runs its own SSH server and its own SSL server, NSO can pick up the password of a user in clear text. Hence, if the 'same-pass' option is chosen for an authgroup, NSO will reuse the same password when attempting to connect southbound to a managed device.
 
+### Connecting Using SSH Keyboard-Interactive (Multi-Factor) Authentication
+
+NSO can connect to a device that is using multi-factor authentication. For this, the `authgroup` must be configured with an executable for handling the keyboard-interactive part, and optionally some opaque data that is passed to the executable. ie., the `/devices/authgroups/group/umap/mfa/executable` and `/devices/authgroups/group/umap/mfa/opaque` (or under `default-map` for users that are not in `umap`) must be configured.
+
+The prompts from the SSH server (including the password prompt and any additional challenge prompts) are passed to the `stdin` of the executable along with some other relevant data. The executable must write a single line to its `stdout` as the reply to the prompt. This is the reply that NSO sends to the SSH server.
+
+{% code title="Example: Configuring Authgroup For Keyboard-interactive Authentication" %}
+```
+admin@ncs(config)# devices authgroups group mfa umap admin
+admin@ncs(config-umap-admin)# remote-name admin remote-password
+(<AES encrypted string>): *********
+admin@ncs(config-umap-admin)# mfa executable ./handle_mfa.py opaque foobar
+admin@ncs(config-umap-admin)# commit
+Commit complete.
+```
+{% endcode %}
+
+For example, with the above configured for the authgroup, if the user _admin_ is trying to login to the device _dev0_ with password _admin_, this is the line that is sent to the `stdin` of the `handle_mfa.py` script:
```

Comment:
This probably is fine in itself, but to keep it consistent (and simple) with other parts of doc, I'd suggest using backticks `` instead of _ _.

```suggestion
For example, with the above configured for the authgroup, if the user `admin` is trying to login to the device `dev0` with password `admin`, this is the line that is sent to the `stdin` of the `handle_mfa.py` script:
```